solr/core/src/java/org/apache/solr/search/Grouping.java

```diff
@@ -461,7 +461,16 @@ private void searchWithTimeLimiter(final Query filterQuery, Collector collector)
       searcher.search(QueryUtils.combineQueryAndFilter(query, filterQuery), collector);
     } catch (TimeLimitingCollector.TimeExceededException
         | ExitableDirectoryReader.ExitingReaderException x) {
-      log.warn("Query: {}; ", query, x);
+      if (log.isWarnEnabled()) {
+        String queryToLog = query.toString();
+        if (queryToLog.length() > 1000) {
```

Comment:
Is this a common enough scenario to justify increasing the code complexity here like this? And if so would 1000 be a reasonable limit?